src/addons/Confirm/Confirm.js

```diff
@@ -1,68 +1,103 @@
-      {content && <Modal.Content>{content}</Modal.Content>}
-      <Modal.Actions>
-        <Button onClick={onCancel}>{cancelButton}</Button>
-        <Button primary onClick={onConfirm}>{confirmButton}</Button>
-      </Modal.Actions>
-    </Modal>
-  )
-}
+class Confirm extends Component {
+  static propTypes = {
+    /** The cancel button text. */
+    cancelButton: PropTypes.string,
 
-Confirm._meta = {
-  name: 'Confirm',
-  type: META.TYPES.ADDON,
-}
+    /** The OK button text. */
+    confirmButton: PropTypes.string,
 
-Confirm.propTypes = {
-  /** Whether or not the modal is visible */
-  open: PropTypes.bool,
+    /** The ModalContent text. */
+    content: customPropTypes.itemShorthand,
 
-  /** The cancel button text */
-  cancelButton: PropTypes.string,
+    /** The ModalHeader text. */
+    header: customPropTypes.itemShorthand,
 
-  /** The OK button text */
-  confirmButton: PropTypes.string,
+    /**
+     * Called when the Cancel button is clicked.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onCancel: PropTypes.func,
 
-  /** The ModalHeader text */
-  header: PropTypes.string,
+    /**
+     * Called when the OK button is clicked.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onConfirm: PropTypes.func,
 
-  /** The ModalContent text. */
-  content: PropTypes.string,
+    /** Whether or not the modal is visible. */
+    open: PropTypes.bool,
+  }
 
-  /** Called when the OK button is clicked */
-  onConfirm: PropTypes.func,
+  static defaultProps = {
+    cancelButton: 'Cancel',
+    confirmButton: 'OK',
+    content: 'Are you sure?',
+  }
 
-  /** Called when the Cancel button is clicked */
-  onCancel: PropTypes.func,
-}
+  static _meta = {
+    name: 'Confirm',
+    type: META.TYPES.ADDON,
+  }
+
+  handleCancel = e => {
+    const { onCancel } = this.props
+
+    if (onCancel) onCancel(e, this.props)
+  }
+
+  handleConfirm = e => {
+    const { onConfirm } = this.props
+
+    if (onConfirm) onConfirm(e, this.props)
+  }
+
+  render() {
+    const {
+      cancelButton,
+      confirmButton,
+      content,
+      header,
+      open,
+    } = this.props
+    const rest = getUnhandledProps(Confirm, this.props)
+
+    // `open` is auto controlled by the Modal
+    // It cannot be present (even undefined) with `defaultOpen`
+    // only apply it if the user provided an open prop
+    const openProp = {}
+    if (_.has(this.props, 'open')) openProp.open = open
 
-Confirm.defaultProps = {
-  cancelButton: 'Cancel',
-  confirmButton: 'OK',
-  content: 'Are you sure?',
+    return (
+      <Modal {...rest} {...openProp} size='small' onClose={this.handleCancel}>
+        {createShorthand(Modal.Header, val => ({ children: val }), header)}
+        {createShorthand(Modal.Content, val => ({ children: val }), content)}
```

Comment:
I'm slowing trying to remove use of `createShorthand` in favor of adding `create()` methods to the component when needed.  We do plan to have shorthand for the Modal, which we currently don't, so having the `create()` methods for ModalHeader and ModalContent would be helpful.

Let's remove `createShorthand` and add the `create()` method for these components instead.